### PR TITLE
Test improvements

### PR DESCRIPTION
**Pull request by jarhodes314**
## Proposed changes
This makes a few test improvements. One of them concerns running the tests under `cargo-nextest`, which fails on some MQTT tests which spin up a server on a random port due to a conflict. The other two are changes to fix the tests on my machine, as they assume I have a group with the same name as my username, and assume a specific error messages for DNS errors (they expect `dns error: Failed to lookup address information`, but my machine gives the error `dns error: Device or resource busy`).

For simplicity, I've assumed there is always a group called `root`, but I think this a more reliable assumption than a group sharing a name with the current username.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/common/tedge_utils/src/file.rs [hunk 1]
```diff
@@ -488,7 +488,7 @@ mod tests {
         let file_path = temp_dir.path().join("file").display().to_string();
 
         let user = whoami::username();
-        create_file_with_user_group(&file_path, &user, &user, 0o644, None).unwrap();
+        create_file_with_user_group(&file_path, &user, "root", 0o644, None).unwrap();
```
Comment (Bravo555) [inline crates/common/tedge_utils/src/file.rs:491]: Why not just use the groupname of the user who launched the process?
```suggestion
        let group = uzers::get_current_groupname().unwrap();
        let group = group.to_str().unwrap();
        create_file_with_user_group(&file_path, &user, group, 0o644, None).unwrap();
```

Comment (jarhodes314) [reply crates/common/tedge_utils/src/file.rs:491]: I didn't realise we already had a dependency that made getting that information easy, I'll change it

File: justfile [hunk 2]
```diff
@@ -62,7 +62,8 @@ release *ARGS:
 
 # Run unit tests
 test:
-    cargo test --no-fail-fast --all-features --all-targets
+    cargo nextest run --no-fail-fast --all-features --all-targets
+    cargo test --doc --no-fail-fast --all-features
```
Comment (didier-wenzek) [inline justfile:66]: Indeed, a trap I fall into: `nextest` skips doc tests.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**